plugins/inputs/github/github.go

```diff
@@ -152,6 +163,35 @@ func (g *GitHub) Gather(acc telegraf.Accumulator) error {
 	return nil
 }
 
+func (g *GitHub) handleRateLimit(response *github.Response, err error) {
+	if err == nil {
+		g.RateLimit.Set(int64(response.Rate.Limit))
+		g.RateRemaining.Set(int64(response.Rate.Remaining))
+	} else if _, ok := err.(*github.RateLimitError); ok {
+		g.RateLimitErrors.Incr(1)
+	}
+}
+
+func (g *GitHub) gatherAdditionalFields(ctx context.Context, owner, repo string) (map[string]interface{}, error) {
+	fields := make(map[string]interface{})
+
+	// Pull request properties
+	options := github.SearchOptions{TextMatch: false, ListOptions: github.ListOptions{PerPage: 1, Page: 1}}
+	classes := []string{"open", "closed"}
+	for _, class := range classes {
+		q := fmt.Sprintf("repo:%s/%s is:pr is:%s", owner, repo, class)
+		searchResult, response, err := g.githubClient.Search.Issues(ctx, q, &options)
```

Comment:
It looks like this makes an additional http round trip to get the PR counts?  This is a cost we should mention in the config file and readme so when users choose to enable this new functionality they understand why they might not want to.

Reply:
Agreed will do.